### PR TITLE
feat: add stringx to lang pkg

### PR DESCRIPTION
**Pull request by kaka19ace**

### CODE REVIEW

**Review by PureWhiteWu — APPROVED** [review]

**Comment by PureWhiteWu** [comment]
加个 README.md 介绍一下？

**Review by zhangyunhao116 — COMMENTED** [review]
Thanks for your PR!  Please add a readme to describe this repo.

File: lang/stringx/is.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+// limitations under the License.
+
+package stringx
+
+import (
+	"unicode"
+)
+
+// IsBlank checks if string is empty ("") or whitespace only.
+func IsBlank(s string) bool {
```
[the remark this answers is not in the document]
Comment (kaka19ace) [reply lang/stringx/is.go:22]: in Java commons-lang3 StringUtils,  blank check  includes whitespace check.  https://github.com/apache/commons-lang/blob/master/src/main/java/org/apache/commons/lang3/StringUtils.java#L3579

File: lang/stringx/case.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+import (
+	"strings"
+	"unicode"
+	"unicode/utf8"
+)
+
+// ToCamelCase Converts all the delimiter separated words in a String into camelCase,
+// that is each word is made up of a titlecase character and then a series of lowercase characters.
+// See https://en.wikipedia.org/wiki/Camel_case
+func ToCamelCase(s string, isUpper bool, delimiters ...rune) string {
```
[the remark this answers is not in the document]
Comment (kaka19ace) [reply lang/stringx/case.go:26]: > Bad case
> `println(strings.ToCamelCase("check_API_framework", false))` -> `check_api_framework`

you should add specified delimiter '_':

File: lang/stringx/stringx.go [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+	}
+	return sb.String()
+}
+
+// RemoveString removes all occurrences of a substring from the string.
+func RemoveString(s, rmStr string) string {
+	if s == "" || rmStr == "" {
+		return s
+	}
+	return strings.ReplaceAll(s, rmStr, "")
```
Comment (zhangyunhao116) [inline lang/stringx/stringx.go:149]: Feel like we could use `strings.ReplaceAll` directly, we don't need to wrap it into a function.

File: lang/stringx/stringx.go [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+			return true
+		}
+	}
+	return false
+}
+
+// Utf8Len returns str's utf8 rune length.
+// Erroneous and short encodings are treated as single runes of width 1 byte.
+func Utf8Len(s string) int {
+	return utf8.RuneCountInString(s)
```
Comment (zhangyunhao116) [inline lang/stringx/stringx.go:290]: same as above, we can use utf8 directly.

File: lang/stringx/stringx.go [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+
+// Shuffle  shuffles runes in a string and returns.
+func Shuffle(s string) string {
+	if s == "" {
+		return s
+	}
+	runes := []rune(s)
+	index := 0
+	for i := len(runes) - 1; i > 0; i-- {
+		index = rand.Intn(i + 1)
```
Comment (zhangyunhao116) [inline lang/stringx/stringx.go:269]: We can use`fastrand`, it is more scalable.

File: lang/stringx/stringx.go (hunk 3 above)
Comment (kaka19ace) [reply lang/stringx/stringx.go:149]: > Feel like we could use `strings.ReplaceAll` directly, we don't need to wrap it into a function.

we provide a clear/breif wrapper is better than using strings.ReplaceAll directly.

File: lang/stringx/stringx.go (hunk 4 above)
Comment (kaka19ace) [reply lang/stringx/stringx.go:290]: > same as above, we can use utf8 directly.

RuneCountInString is not a straight name,
and it's not a nonessential feature, i'll disable it temporarily.  When if there is actual need from feedback in future,  We can start the discussion again.

File: lang/stringx/stringx.go (hunk 5 above)
Comment (kaka19ace) [reply lang/stringx/stringx.go:269]: sounds good

**Review by zhangyunhao116 — APPROVED** [review]
Thanks!

**Review by PureWhiteWu — APPROVED** [review]

**Pull request merged**